### ISSUE TITLE
An Error occurred in single-UAV 3D object detection test time (DETR3D) 

### ISSUE THREAD

**Issue by tdd233** [huiyegit/UAV3D#9]
Hi~Thanks for your outstanding work! When I tried to reproduce your method, an error occurred in test time as follows:
![image](https://github.com/user-attachments/assets/3d48e6c7-04fd-4699-938d-634ab28b94d0)
It seems like the data loaded from prediction and gt is different? Could you please tell me how to solve the problem?
Looking forward to your reply!!!

**Comment by huiyegit** [comment]
Please verify the dataset version: UAV3D or UAV3D-mini.

**Comment by tdd233** [comment]
> Please verify the dataset version: UAV3D or UAV3D-mini.

Thanks for your prompt reply. I have verified the dataset version and there is no problem. But I found that the prediction bbox is from test set and the gt bbox is from validation set which caused the error. Could you please tell me where to change the data loading path of gt bbox? 

Looking forward to your reply.

**Comment by huiyegit** [comment]
Please check the config file: https://github.com/huiyegit/UAV3D/blob/main/perception/detr3d/projects/configs/detr3d/detr3d_res101_gridmask.py

<img width="1004" alt="Screen Shot 2024-12-16 at 3 52 21 AM" src="https://github.com/user-attachments/assets/a8ce4ab7-91ec-4faa-aaf4-97d3c55241e8" />

**Comment by clever-mood** [comment]
> > Please verify the dataset version: UAV3D or UAV3D-mini.
> 
> Thanks for your prompt reply. I have verified the dataset version and there is no problem. But I found that the prediction bbox is from test set and the gt bbox is from validation set which caused the error. Could you please tell me where to change the data loading path of gt bbox?
> 
> Looking forward to your reply.

hi，I meet the same question，do you solve it？

**Comment by Layort** [comment]
> > Please verify the dataset version: UAV3D or UAV3D-mini.
> 
> Thanks for your prompt reply. I have verified the dataset version and there is no problem. But I found that the prediction bbox is from test set and the gt bbox is from validation set which caused the error. Could you please tell me where to change the data loading path of gt bbox?
> 
> Looking forward to your reply.

hi，I meet this problem too bro， i wanna how did you solve it？

**Comment by Layort** [comment]
> Please check the config file: https://github.com/huiyegit/UAV3D/blob/main/perception/detr3d/projects/configs/detr3d/detr3d_res101_gridmask.py
> 
> <img alt="Screen Shot 2024-12-16 at 3 52 21 AM" width="1004" src="https://private-user-images.githubusercontent.com/31486250/396026409-a8ce4ab7-91ec-4faa-aaf4-97d3c55241e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA4MDU1MjgsIm5iZiI6MTc0MDgwNTIyOCwicGF0aCI6Ii8zMTQ4NjI1MC8zOTYwMjY0MDktYThjZTRhYjctOTFlYy00ZmFhLWFhZjQtOTdkM2M1NTI0MWU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzAxVDA1MDAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5ZWM0MzQyYjljZDZmMzlkZjQ5N2RiNTc3NzNiYWU1MzI1MGE4YjE0ZTcxMThjMjM4MDU5NGM1NTQyMzEwZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2IfPQRcjbsWd-kiDXD16I2GU_UTRP_a5mXPr9mWEXBk">

I just opened the "uav3d_infos_test.pkl" file and found that it's version info names "'v1.0-trainval'". Is that right?

**Comment by icefrogplus** [comment]
just change the code ".pkl"

```python
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=4,
    train=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file=data_root + 'uav3d_infos_train.pkl',
        pipeline=train_pipeline,
        classes=class_names,
        modality=input_modality,
        test_mode=False,
        use_valid_flag=True,
        # we use box_type_3d='LiDAR' in kitti and nuscenes dataset
        # and box_type_3d='Depth' in sunrgbd and scannet dataset.
        box_type_3d='LiDAR'),
    val=dict(pipeline=test_pipeline, data_root=data_root, ann_file=data_root + 'uav3d_infos_val.pkl', classes=class_names, modality=input_modality),
    test=dict(pipeline=test_pipeline, data_root=data_root, ann_file=data_root + 'uav3d_infos_test.pkl',  classes=class_names, modality=input_modality))
```

the code just for train stage to match the original Nuscenes dataset
```python
test=dict(pipeline=test_pipeline, data_root=data_root, ann_file=data_root + 'uav3d_infos_test.pkl',  classes=class_names,
```

but when val, we need to change the val data
```python
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=4,
    train=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file=data_root + 'uav3d_infos_train.pkl',
        pipeline=train_pipeline,
        classes=class_names,
        modality=input_modality,
        test_mode=False,
        use_valid_flag=True,
        # we use box_type_3d='LiDAR' in kitti and nuscenes dataset
        # and box_type_3d='Depth' in sunrgbd and scannet dataset.
        box_type_3d='LiDAR'),
    val=dict(pipeline=test_pipeline, data_root=data_root, ann_file=data_root + 'uav3d_infos_val.pkl', classes=class_names, modality=input_modality),
    test=dict(pipeline=test_pipeline, data_root=data_root, ann_file=data_root + 'uav3d_infos_val.pkl',  classes=class_names, modality=input_modality))
```

then, here is the val result
```txt
Evaluating bboxes of pts_bbox
mAP: 0.0614                                                                                                                                                                                                       
mATE: 0.9503
mASE: 0.9157
mAOE: 0.8968
mAVE: 1.0000
mAAE: 1.0000
NDS: 0.0544
Eval time: 534.4s

Per-class results:
Object Class    AP      ATE     ASE     AOE     AVE     AAE
car     0.614   0.503   0.157   0.071   1.000   1.000
truck   0.000   1.000   1.000   1.000   1.000   1.000
bus     0.000   1.000   1.000   1.000   1.000   1.000
trailer 0.000   1.000   1.000   1.000   1.000   1.000
construction_vehicle    0.000   1.000   1.000   1.000   1.000   1.000
pedestrian      0.000   1.000   1.000   1.000   1.000   1.000
motorcycle      0.000   1.000   1.000   1.000   1.000   1.000
bicycle 0.000   1.000   1.000   1.000   1.000   1.000
traffic_cone    0.000   1.000   1.000   nan     nan     nan
barrier 0.000   1.000   1.000   1.000   nan     nan
{'pts_bbox_NuScenes/car_AP_dist_0.5': 0.2241, 'pts_bbox_NuScenes/car_AP_dist_1.0': 0.5371, 'pts_bbox_NuScenes/car_AP_dist_2.0': 0.7869, 'pts_bbox_NuScenes/car_AP_dist_4.0': 0.9073, 'pts_bbox_NuScenes/car_trans_err': 0.5025, 'pts_bbox_NuScenes/car_scale_err': 0.1572, 'pts_bbox_NuScenes/car_orient_err': 0.0709, 'pts_bbox_NuScenes/car_vel_err': 1.0, 'pts_bbox_NuScenes/car_attr_err': 1.0, 'pts_bbox_NuScenes/mATE': 0.9503, 'pts_bbox_NuScenes/mASE': 0.9157, 'pts_bbox_NuScenes/mAOE': 0.8968, 'pts_bbox_NuScenes/mAVE': 1.0, 'pts_bbox_NuScenes/mAAE': 1.0, 'pts_bbox_NuScenes/NDS': 0.05441901951620702, 'pts_bbox_NuScenes/mAP': 0.061384470285205064}
